### DISCUSSION TITLE
Not able to load new namespace dynamically at client-side

### DISCUSSION THREAD

**Discussion by mikaelnguyenpg** [i18next/next-i18next#2307] [Q&A]
Hi guys, I did try the sample SSG of this library
My goal is to test whether can load a namespace dynamically at client-side
Hence I make new json file `test.json` in each lang of locales, and extract key-value `description` from `footer.json` into `test.json`
I think that, in theory, in component Footer, i use `const { t } = useTranslation(['footer', 'test'])`, it should be able to load the new namespace `test` fine and the text of key `description` should be displayed, isn't it?
but result is that not able to display
![image](https://github.com/user-attachments/assets/6f7e441a-7aab-4adb-b7c4-e8f85f347aee)
the code i use is this
![image](https://github.com/user-attachments/assets/eeb180f6-335c-46ea-8494-1d820a683c29)
Could anyone explain for me why? and how to fix?
Thanks

**Comment by adrai** [comment]
If you can't add it on server side: https://github.com/i18next/next-i18next?tab=readme-ov-file#5-declaring-locale-dependencies

You need to load them like this: https://github.com/i18next/next-i18next?tab=readme-ov-file#client-side-loading-of-translations-via-http => https://github.com/i18next/i18next-http-backend/tree/master/example/next
